src/Dumplie/Domain/Customer/Product.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Domain\Customer;
+
+use Dumplie\Domain\Customer\Exception\InvalidArgumentException;
+use Dumplie\Domain\SharedKernel\Money\Price;
+
+final class Product
+{
+    /**
+     * @var string
+     */
+    private $sku;
+
+    /**
+     * @var Price
+     */
+    private $price;
+
+    /**
+     * @var bool
+     */
+    private $available;
```

Comment:
Thats all true but in the different context. Customer context (and this is where adding to basket command lives) should not be aware of tha. From customer point of view product can be available or not (the only exception is custom adding to basket workflow, just like the one described by @potfur). 

> Amazon allows you to buy unavailable products but will request payment only when they become available. 

Thats good point however adding support for this kind of case sounds like a overkill at least for first release.